initrd/etc/luks-functions

```diff
@@ -272,19 +272,17 @@ select_luks_container()
 		DEBUG "Reusing known good LUKS container device from /boot/kexec_key_devices.txt"
 		DEBUG "LUKS container device: $(cut -d ' ' -f1 /boot/kexec_key_devices.txt)"
 		LUKS=$(cut -d ' ' -f1 /boot/kexec_key_devices.txt)
-	else
+	# LUKS variable not exported yet, prompt for LUKS device
+	elif [ -z $LUKS ]; then
 		list_luks_devices > /tmp/luks_devices.txt
 		#if /tmp/luks_devices.txt exists and is not empty
 		if [ -s /tmp/luks_devices.txt ]; then
 			file_selector "/tmp/luks_devices.txt" "Select LUKS container device"
 			if [ "$FILE" == "" ]; then
 				return 1
 			else
+				#TODO: What about BRTFS multi LUKS setup of QubesOS?
 				LUKS=$FILE
-				detect_boot_device
-				mount -o remount,rw /boot
-				echo "$LUKS $(cryptsetup luksUUID $LUKS)" >/boot/kexec_key_devices.txt
```

Comment:
/boot/kexec_key_devices.txt should only exist if a DUK key is setuped.

Inwillnreciew code once more after addressing your review.

It contains uuids to be used to construct secret.cpio to be appended to initramfs when DUK is to be released to the OS through kexec call.

It was a bug to use it for anything else.

This is why it was removed outside of DUK code path

Reply:
@UndeadDevel this should be addressed, please validate.

Reply:
It looks like `kexec-save-key` writes to kexec_key_devices.txt, I think.

That'd make sense, because it makes more sense to collect all the information we need, then execute and write everything, rather than writing one piece of information at a time.  The latter can leave you in confusing partially-configured states when something gets canceled (which definitely happens from time to time for users, and is hard to diagnose after the fact).

That's just from a preliminary search though, I think this needs to be read through in more detail.  I'm going to go over the rest of the PR first.